File: src/main/java/de/weimarnetz/registrator/services/LinkService.java
package de.weimarnetz.registrator.services;

import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;

import java.net.URI;

import org.springframework.stereotype.Component;

import de.weimarnetz.registrator.controller.RegistratorController;


@Component
public class LinkService {

    public String getNodeLocation(String network, int nodeNumber) {
        return linkTo(methodOn(RegistratorController.class).getSingleNode(network, nodeNumber)).toUri().getPath();
    }

    public URI getNodeLocationUri(String network, int nodeNumber) {
        return linkTo(methodOn(RegistratorController.class).getSingleNode(network, nodeNumber)).toUri();
    }
}

File: src/main/java/de/weimarnetz/registrator/services/LinkService.kt
package de.weimarnetz.registrator.services

import de.weimarnetz.registrator.controller.RegistratorController
import org.springframework.hateoas.server.mvc.linkTo
import org.springframework.stereotype.Component
import java.net.URI

@Component
class LinkService {
    fun getNodeLocation(network: String, nodeNumber: Int): String {
        return linkTo<RegistratorController> { getSingleNode(network, nodeNumber) }.toUri().path
    }

    fun getNodeLocationUri(network: String, nodeNumber: Int): URI {
        return linkTo<RegistratorController> { getSingleNode(network, nodeNumber) }.toUri()
    }
}